phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -2455,6 +2409,19 @@ public Void call() throws Exception {
                                 }
 
                                 if (currentServerSideTableTimeStamp < MetaDataProtocol.MIN_SYSTEM_TABLE_TIMESTAMP_4_8_0) {
+                                    Properties props = PropertiesUtil.deepCopy(metaConnection.getClientInfo());
+                                    props.remove(PhoenixRuntime.CURRENT_SCN_ATTRIB);
+                                    props.remove(PhoenixRuntime.TENANT_ID_ATTRIB);
+                                    PhoenixConnection conn =
+                                            new PhoenixConnection(ConnectionQueryServicesImpl.this,
+                                                    metaConnection.getURL(), props, metaConnection
+                                                            .getMetaDataCache());
+                                    try {
+                                        UpgradeUtil.upgradeLocalIndexes(conn);
```

Comment:
Should we only disable all local indexes here as a full rebuild would take a lot of time? Then we can let users use your utility to rebuild them?